src/components/OnboardingComponents/SemanticSearch/SemanticSearchContainer.tsx

```diff
@@ -0,0 +1,152 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR)
+// SPDX-License-Identifier: Apache-2.0
+
+import React, {useState} from 'react';
+import {Box, Typography, Divider, TextField, IconButton, InputAdornment, CircularProgress} from '@mui/material';
+import {useSemanticSearch} from 'context/SemanticSearchContext';
+import QuerySuggestions from './QuerySuggestions';
+import {useTranslation} from 'react-i18next';
+import SearchIcon from '@mui/icons-material/Search';
+import {useAppDispatch, useAppSelector} from 'store/hooks';
+import {setSemanticSearchQuery} from 'store/SemanticSearchSlice';
+import SearchResultsList from './SearchResultsList';
+import ArticleDialog from './ArticleDialog';
+import {SearchResult} from 'types/semanticSearch';
+
+/**
+ * Main container for the Semantic Search feature.
+ * It will manage the search state and orchestrate the child components.
+ */
+export default function SemanticSearchContainer(): JSX.Element {
+  const {searchResults, isLoading, performSearch, clearSearch} = useSemanticSearch();
+  const dispatch = useAppDispatch();
+  const query = useAppSelector((state) => state.semanticSearch.searchQuery);
+  const searchStatus = useAppSelector((state) => state.semanticSearch.searchStatus);
+  const {t} = useTranslation('global');
+  const [selectedArticle, setSelectedArticle] = useState<SearchResult | null>(null);
+
+  const suggestions = [t('semanticSearch.suggestions.scenarioForecasting'), t('semanticSearch.suggestions.purpose')];
+
+  const handleSearch = () => {
+    // performSearch is now required and will not be undefined.
+    if (performSearch) {
+      performSearch(query);
+    }
+  };
+
+  const handleClear = () => {
+    dispatch(setSemanticSearchQuery(''));
+    clearSearch();
+  };
+
+  const handleSuggestionClick = (suggestion: string) => {
+    dispatch(setSemanticSearchQuery(suggestion));
+    if (performSearch) {
+      performSearch(suggestion);
+    }
+  };
+
+  const handleResultClick = (article: SearchResult) => {
+    setSelectedArticle(article);
+  };
+
+  const handleCloseDialog = () => {
+    setSelectedArticle(null);
+  };
+
+  return (
+    <Box mt={4}>
```

Comment:
same as in the other files. Use sx, system props deprecated.
```suggestion
    <Box
      sx= {{
        marginTop: 4,
      }}
    >
```

Reply:
ok